applications/MeshMovingApplication/custom_processes/impose_mesh_motion_process.cpp

```diff
@@ -191,13 +203,17 @@ void ImposeMeshMotionProcess::ExecuteInitializeSolutionStep()
 {
     KRATOS_TRY;
 
-    block_for_each(mrModelPart.Nodes(),
-        [this](Node<3>& rNode) {
-            array_1d<double,3> transformed_point = rNode;
-            this->Transform(transformed_point);
-            rNode.SetValue(MESH_DISPLACEMENT, transformed_point - rNode);
-        }
-    );
+    const double time = mrModelPart.GetProcessInfo().GetValue(TIME);
+
+    if (mInterval[0] <= time && time <= mInterval[1]) {
```

Comment:
Can you please use the [IntervalUtility](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/utilities/interval_utility.h) here?